tests/run-make/reproducible-build/rmake.rs

```diff
@@ -0,0 +1,170 @@
+// This test case makes sure that two identical invocations of the compiler
+// (i.e. same code base, same compile-flags, same compiler-versions, etc.)
+// produce the same output. In the past, symbol names of monomorphized functions
+// were not deterministic (which we want to avoid).
+//
+// The test tries to exercise as many different paths into symbol name
+// generation as possible:
+//
+// - regular functions
+// - generic functions
+// - methods
+// - statics
+// - closures
+// - enum variant constructors
+// - tuple struct constructors
+// - drop glue
+// - FnOnce adapters
+// - Trait object shims
+// - Fn Pointer shims
+// See https://github.com/rust-lang/rust/pull/32293
+
+// FIXME(Oneirical): ignore-musl
+// FIXME(Oneirical): two of these test blocks will apparently fail on windows
+// FIXME(Oneirical): try it on test-various
+// # TODO: Builds of `bin` crate types are not deterministic with debuginfo=2 on
+// # Windows.
+// # See: https://github.com/rust-lang/rust/pull/87320#issuecomment-920105533
+// # Issue: https://github.com/rust-lang/rust/issues/88982
+
+use run_make_support::{bin_name, cwd, diff, rfs, rust_lib_name, rustc};
+
+fn main() {
+    rustc().input("linker.rs").opt().run();
+    rustc().input("reproducible-build-aux.rs").run();
+    eprintln!("{:#?}", rfs::shallow_find_dir_entries(cwd()));
+    rustc().input("reproducible-build.rs").linker(&bin_name("linker")).run();
+    rustc().input("reproducible-build.rs").linker(&bin_name("linker")).run();
+    diff().actual_file("linker-arguments1").expected_file("linker-arguments2").run();
+    rustc().input("linker.rs").opt().run();
+    rustc().arg("-g").input("reproducible-build-aux.rs").run();
+    rustc().arg("-g").input("reproducible-build.rs").linker(&bin_name("linker")).run();
+    rustc().arg("-g").input("reproducible-build.rs").linker(&bin_name("linker")).run();
+    diff().actual_file("linker-arguments1").expected_file("linker-arguments2").run();
+    rustc().input("linker.rs").opt().run();
+    rustc().opt().input("reproducible-build-aux.rs").run();
+    rustc().opt().input("reproducible-build.rs").linker(&bin_name("linker")).run();
+    rustc().opt().input("reproducible-build.rs").linker(&bin_name("linker")).run();
```

Comment:
Problem: these `-Clinker=<path>` flags need to use either `./linker` or `cwd().join("linker")` (well, with the `bin_name` as suitable) because eventually codegen_ssa uses `Command::new()` to do linker detection, and if a relative path is used that's not `./` prefixed it starts searching in `PATH`.